src/net/rpc.rs

```diff
@@ -1532,23 +1531,75 @@ impl ConversationHttp {
     /// Load up the canonical Stacks chain tip.  Note that this is subject to both burn chain block
     /// Stacks block availability -- different nodes with different partial replicas of the Stacks chain state
     /// will return different values here.
-    /// tip_opt is given by the HTTP request as the optional query parameter for the chain tip
-    /// hash.  It will be None if there was no paramter given.
-    /// The order of chain tips this method prefers is as follows:
-    /// * tip_opt, if it's Some(..),
-    /// * the unconfirmed canonical stacks chain tip, if initialized
-    /// * the confirmed canonical stacks chain tip
+    ///
+    /// # Inputs
+    /// - `tip_req` is given by the HTTP request as the optional query parameter for the chain tip
+    /// hash.  It will be UseLatestAnchoredTip if there was no parameter given. If it is set to
+    /// `latest`, the parameter will be set to UseLatestUnconfirmedTip.
     fn handle_load_stacks_chain_tip<W: Write>(
         http: &mut StacksHttp,
         fd: &mut W,
         req: &HttpRequestType,
-        tip_opt: Option<&StacksBlockId>,
+        tip_req: &TipRequest,
         sortdb: &SortitionDB,
-        chainstate: &StacksChainState,
+        chainstate: &mut StacksChainState,
     ) -> Result<Option<StacksBlockId>, net_error> {
-        match tip_opt {
-            Some(tip) => Ok(Some(*tip).clone()),
-            None => match chainstate.get_stacks_chain_tip(sortdb)? {
+        match tip_req {
+            TipRequest::UseLatestUnconfirmedTip => {
+                let unconfirmed_chain_tip_opt = match &mut chainstate.unconfirmed_state {
+                    Some(unconfirmed_state) => {
```

Comment:
Minor refactoring nit: can you add a method to `UnconfirmedState` called `get_unconfirmed_tip_if_exists()` that does all this work here?  Then, if it fails -- i.e. the underlying trie doesn't exist -- this code block would simply return the appropriate HTTP response type.